### ISSUE TITLE
Server nicht verfügbar nach Update vom 18.11.2020

### ISSUE THREAD

**Issue by KlausThomasMueller** [nextcloud/android#7444]
### Steps to reproduce
1. Open App and look on "Uploads"
2. Problem is on both of my mobile phones and with https://try.nextcloud.com
3. Windows Laptop works fine

### Expected behaviour
- Files should be uploaded

### Actual behaviour
- Message "Verbindungsfehler" is shown and nothing is uploaded. 

### Can you reproduce this problem on https://try.nextcloud.com?
- Please create a test demo account and see if this still happens there.
- If yes, please open up a bug report
- If not, please verify server setup and ask for help on forum

### Environment data
Android version:
8.0.0

Device model: 
Samsung SM A520F 

Stock or customized system:
Stock

Nextcloud app version:
3.14.0

Nextcloud server version:

Reverse proxy:

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by ftastisch** [comment]
same here

**Comment by AndyScherzinger** [comment]
Which server version of Nextcloud are you using?

**Comment by KlausThomasMueller** [comment]
Nextcloud 19.0.2 on Linux Mint 20 Ulyana

**Comment by ftastisch** [comment]
Nextcloud 20.0.0 on Manjaro.

**Comment by CarlosCastaneda** [comment]
nexcloud 

  | 19.0.5.2
-- | --



  | 19.0.5.2
Nextcloud 19.0.5.2 auf Raspberry Pi

**Comment by xtype307sw** [comment]
I can confirm the problem.
NC 19.0.4 on RasPi4.
Access from the local LAN is disturbed with the Android app 3.14. Outside access is OK. At the same time there are no problems with the IOS app, the Windows app or with browsers.

**Comment by ftastisch** [comment]
My instance is also running on a Raspi 4 and I can confirm access when not in the same network.

**Comment by tobiasKaminsky** [comment]
Is this about file browsing or auto-upload? 
It seems that there are different reports.

Also it seems to be more a problem when device is in the same wlan with server?

**Comment by KlausThomasMueller** [comment]
When I create a new account in the APP within my local WLAN and with my internal IP (https://192.168.178.xx/nextcloud), it is possible to connect to the server. It is also possible to upload and download files.
When I use my external connection, as usual, https://aqxxxxxxxxxx.myfritz.net/nextcloud, I get the message "Server not found" when trying to create a new account (at the input screen for the server address), and the uploads for the existing account over these connection stop with a "Connection error"!
Connecting with the WEB-Interface works from internal and from external.
Additional information, the SSL-certificate is signed only internally but not from a service provider.
Everything was OK until the update last Friday.

**Comment by CarlosCastaneda** [comment]
I also can confirm the very same behaviour seen by KlausThomasMueller.

Furthermore I tested the remote connections (xxxx.myfritz.net) without FritzBox WLAN using iPhones Hotspot: no connection via Nexcloud App - neither existing (e.g. "Laden des Dokuments fehlgeschlagen" ) nor adding a new one ("Verbindung konnte nicht hergestellt werden" ). In crontrast connections may be established sucessfully running e.g. FireFox.

Local connections e.g. https://nextcloudpi within LAN work fine.

**Comment by kurzschlussidi** [comment]
I can also confirm the same problem on two android smartphones running the App version 3.14.0.
When connected to the Wi-Fi or even a LAN-Cable of the fritzbox i get the message "Server nicht verfügbar" when browsing the files and I also get upload errors with the automatic picture upload.
When using mobile Data the app works fine and when connecting to the wifi hotspot of a win 10 pc (also connected with the fritzbox) the app is working fine.
I'm using a public domain with c-name forwarding onto a myfritz.net address.
Connection to the server over the browser is always working.

**Comment by bllngr** [comment]
I can confirm the issue on 3.14.0 and 3.14.1 RC 1 with server version 19.0.5. I had quite some issues with logging in, auto-upload and general stability of the app. I've reverted to 3.13.1 for now, and disabled auto-updates in GPlay until the issue is fixed.

**Comment by torti42** [comment]
Same problem on my Android 10 device - App version 3.14.0 is unusable. Just shows "server not available". Accessing the Nextcloud web interface via Firefox from Android works just fine.

**Comment by tobiasKaminsky** [comment]
I could only imagine that https://github.com/nextcloud/android-library/pull/503 might cause trouble.
But this is about our new internal client, which is not used that often right now.

Can you provide us additional infos via logcat? https://github.com/nextcloud/android/blob/master/README.md#getting-debug-info-via-logcat

**Comment by torti42** [comment]
Thanks, Tobias! Looking into android-library#503 brought me to the root cause: My dynamic DNS included an IPv6 address as well as an IPv4 address. Unfortunately, my Nextcloud server is not reachable via this IPv6 address. Because the new version of the network library prefers IPv6 over IPv4 if available from the DNS name, my server is not reachable.

When I wiped out the IPv6 address from my dynamic DNS record, the App worked again.

**Comment by ftastisch** [comment]
Same here, adding the IPv6 address to my DNS service solved the connection issue.

**Comment by CarlosCastaneda** [comment]
Just disabled FritzBox' IPv6 support and now Nexcloud App connects over xxxx.myfritz.net connetion setting.
Will this be changed in the App so that IPv6 support may be activated again?

**Comment by tobiasKaminsky** [comment]
> Will this be changed in the App so that IPv6 support may be activated again?

I do not fully understand the question.
The app now prefers ipV6 over ipV4. 
If V6 is available, but not properly configured it will fail. Same if you only have V4, but also not properly configured.

**Comment by CarlosCastaneda** [comment]
Now I am a little confused. I had ipV4 and ipV6 configured and the app worked fine until version 3.14. Now I have deactivated ipV6 and it works again, but not with ipV6. I had not changed my settings. The problem exists since the last update of the app.

**Comment by daemonkeeper** [comment]
So I've run into this as well, and was almost certain that this would be an issue in the Nextcloud client. 

I can confirm this is a non-issue (besides of broken IPv6 client setups). The error handling is not very helpful either, but the issue is definitely a client side network protocol problem. If you run into this problem the most likely use case is, that you're having an IPv6 address configured on your phone, but IP forwarding with IPv6 to be not actually working in conbination with an Nextcloud server which does have AAAA pointers set. This may happen if your router has SLAAC enabled, but your uplink connectivity is not allwoing IPv6 forwarding, or - in my case - my firewall not allowing IPv6 forwarding into the WAN in my policy routing.

The simplest way to find out if you have a broken IPv6 is to simply disable WiFi, and use your phone's mobile data connection (e.g. 4G LTE). I did so as a quick test, and discovered that the issue disappears there in the nextcloud client. thus, my journey started. 

tl;dr: disable IPv6 or fix your IPv6 routing setup and this problem disappears. I'm not sure this needs a software fix, but perhaps you might want to check if IPv6 actually works before falling back to it.

**Comment by AndyScherzinger** [comment]
> Now I am a little confused. I had ipV4 and ipV6 configured and the app worked fine until version 3.14. Now I have deactivated ipV6 and it works again, but not with ipV6. I had not changed my settings. The problem exists since the last update of the app.

That simply means that we preferred ipv4 before the last update (setup worked for you ipv6 never got used by the app), no ipv6 is preferred which fails in your case leading to the conclusion that your ipv6 setup is having an issue (not the app!)

**Comment by kurzschlussidi** [comment]
Thank you!
Solved it for me too.
For whatever reason, the ipv6 address in the fritzbox was different than the ipv6 address the server reported
After changing the fritzbox side to match with the server it was working.

**Comment by CarlosCastaneda** [comment]
> @CarlosCastaneda That simply means that we preferred ipv4 before the last update (setup worked for you ipv6 never got used by the app), now ipv6 is preferred which fails in your case leading to the conclusion that your ipv6 setup is having an issue (not the app!)

Thanks for your reply, also @daemonkeeper, in fact, it was my ipv6 setup in the fritzbox which was due to my low or incomplete ipv6 knowledge.  Now that I've dealt with it, I finally managed to make the connection with the help of the link  [https://steemit.com/deutsch/@gldhnchn/03-raspberry-pi-mit-ds-lite-aus-dem-internet-erreichbar-teil-1-ipv6](https://steemit.com/deutsch/@gldhnchn/03-raspberry-pi-mit-ds-lite-aus-dem-internet-erreichbar-teil-1-ipv6).

**Comment by tobiasKaminsky** [comment]
Glad to hear that you all worked it out :-)

**Comment by KlausThomasMueller** [comment]
My topics are nearly solved as well. IPV6 connection works when I connect with a new account.
This means user_mobile@**server**.xyz.myfritz.net/nextcloud (IPV6) works but the still pending uploads on user_mobile@xyz.myfritz.net/nextcloud (released port 443 for IPV4) can not be changed to the new connection. So I will loose the uploads between the 18.11. and yesterday. The connection should be changeable for uploads!

**Comment by KlausThomasMueller** [comment]
Thanks to everybody for all the good hints to solve the topic. :-)

**Comment by redtux** [comment]
Hi everybody, sorry for commenting on an old issue, but the "Server nicht verfügbar" issue still remains for me:

- NC Server 22.2 (hosted by Hetzner)
- NC Android App 3.18.0 (from Google Play store, also tested with F-Droid release)
- Android 11 on Samsung A52 with One UI 3.1 (last updated Dec 2, 2021)
- not using IPv6 (IPv4 only - both on the phone and in the wifi network)
- Nextcloud Security Scan reports no issues…

Please tell me if you need more information in order to debug this issue (e.g. debug info from logcat).
